src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * citus_depended_object.c
+ *
+ * Implements exposed functions related to hiding citus depended objects.
+ *
+ * Copyright (c) Citus Data, Inc.
+ */
+
+#include "postgres.h"
+#include "miscadmin.h"
+
+#include "catalog/namespace.h"
+#include "catalog/pg_aggregate.h"
+#include "catalog/pg_am.h"
+#include "catalog/pg_attribute.h"
+#include "catalog/pg_attrdef.h"
+#include "catalog/pg_constraint.h"
+#include "catalog/pg_class.h"
+#include "catalog/pg_depend.h"
+#include "catalog/pg_enum.h"
+#include "catalog/pg_event_trigger.h"
+#include "catalog/pg_language.h"
+#include "catalog/pg_namespace.h"
+#include "catalog/pg_proc.h"
+#include "catalog/pg_rewrite.h"
+#include "catalog/pg_sequence.h"
+#include "catalog/pg_statistic.h"
+#include "catalog/pg_trigger.h"
+#include "catalog/pg_ts_config.h"
+#include "catalog/pg_ts_dict.h"
+#include "catalog/pg_ts_template.h"
+#include "catalog/pg_type.h"
+#include "distributed/citus_depended_object.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/listutils.h"
+#include "nodes/makefuncs.h"
+#include "nodes/nodeFuncs.h"
+#include "nodes/parsenodes.h"
+#include "utils/lsyscache.h"
+
+/*
+ * GUC hides any objects, which depends on citus extension, from pg meta class queries,
+ * it is intended to be used in vanilla tests to not break postgres test logs
+ */
+bool HideCitusDependentObjects = false;
+
+static Node * CreateCitusDependentObjectExpr(int pgMetaTableVarno, int pgMetaTableOid);
+static List * GetCitusDependedObjectArgs(int pgMetaTableVarno, int pgMetaTableOid);
+
+/*
+ * IsPgLocksTable returns true if RTE is pg_locks table.
+ */
+bool
+IsPgLocksTable(RangeTblEntry *rte)
+{
+	Oid pgLocksId = get_relname_relid("pg_locks", get_namespace_oid("pg_catalog", false));
+	return rte->relid == pgLocksId;
+}
+
+
+/*
+ * SetLocalHideCitusDependentObjectsDisabledWhenAlreadyEnabled disables the GUC HideCitusDependentObjects
+ * if only it is enabled for local transaction.
+ */
+void
+SetLocalHideCitusDependentObjectsDisabledWhenAlreadyEnabled(void)
+{
+	if (!HideCitusDependentObjects)
+	{
+		return;
+	}
+
+	set_config_option("citus.hide_citus_dependent_objects", "false",
+					  (superuser() ? PGC_SUSET : PGC_USERSET), PGC_S_SESSION,
+					  GUC_ACTION_LOCAL, true, 0, false);
+}
+
+
+/*
+ * HideCitusDependentObjectsOnQueriesOfPgMetaTables adds a NOT is_citus_depended_object(oid, oid) expr
+ * to the quals of meta class RTEs that we are interested in.
+ */
+bool
+HideCitusDependentObjectsOnQueriesOfPgMetaTables(Node *node, void *context)
+{
+	if (!CitusHasBeenLoaded() || !HideCitusDependentObjects || node == NULL)
+	{
+		return false;
+	}
+
+	/* prevent calling the function when HideCitusDependentObjects is disabled. */
```

Comment:
As we already return when !HideCitusDependentObjects above, this assert becomes useless.

Can you move this as the first statement of this function?